packages/elements/src/component-factory-strategy.ts

```
@@ -0,0 +1,244 @@
+    const projectableNodes =
+        extractProjectableNodes(element, this.componentFactory.ngContentSelectors);
+    this.componentRef = this.componentFactory.create(childInjector, projectableNodes, element);
+
+    this.implementsOnChanges =
+        isFunction((this.componentRef.instance as any as OnChanges).ngOnChanges);
+
+    this.initializeInputs();
+    this.initializeOutputs();
+
+    this.detectChanges();
+
+    const applicationRef = this.injector.get<ApplicationRef>(ApplicationRef);
+    applicationRef.attachView(this.componentRef.hostView);
+  }
+
+  /** Set any stored initial inputs on the component's properties. */
+  protected initializeInputs(): void {
+    this.componentFactory.inputs.forEach(({propName}) => {
+      const initialValue = this.initialInputValues.get(propName);
+      if (initialValue) {
+        this.setPropertyValue(propName, initialValue);
+      } else {
+        // Keep track of inputs that were not initialized in case we need to know this for
+        // calling ngOnChanges with SimpleChanges
+        this.uninitializedInputs.add(propName);
+      }
+    });
+
+    this.initialInputValues.clear();
+  }
+
+  /** Sets up listeners for the component's outputs so that the events stream emits the events. */
+  protected initializeOutputs(): void {
+    const eventEmitters = this.componentFactory.outputs.map(({propName, templateName}) => {
+      const emitter = (this.componentRef !.instance as any)[propName] as EventEmitter<any>;
+      return map.call(emitter, (value: any) => ({name: templateName, value}));
+    });
+
+    this.events = merge(...eventEmitters);
+  }
+
+  /** Calls ngOnChanges with all the inputs that have changed since the last call. */
+  protected callNgOnChanges(): void {
+    if (!this.implementsOnChanges || this.inputChanges === null) {
+      return;
+    }
+
+    (this.componentRef !.instance as any as OnChanges).ngOnChanges(this.inputChanges);
+    this.inputChanges = null;
```

I think it is better to set `inputChanges` to null _before_ calling `ngOnChanges()` (see [here][1]), because now you will silently ignore any new changes that happen during `ngOnChanges()`. 

[1]: https://github.com/angular/angular/pull/21106/files#diff-b9835a8f3bd5303fb28263190a46af84R256